From 25e702ca3aa30cceb49a9961483a043015ecbc66 Mon Sep 17 00:00:00 2001
From: Alexander Sack <asac@pantacor.com>
Date: Mon, 22 Apr 2024 20:21:43 +0000
Subject: [PATCH] fix conflicting boot.scr from of u-boot and
 u-boot-default-script recipe.

Do not add u-boot-default-script to DEPENDS if UBOOT_ENV is set.
---
 recipes-bsp/u-boot/u-boot_%.bbappend | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/recipes-bsp/u-boot/u-boot_%.bbappend b/recipes-bsp/u-boot/u-boot_%.bbappend
index e50acf5..0f29bc3 100644
--- a/recipes-bsp/u-boot/u-boot_%.bbappend
+++ b/recipes-bsp/u-boot/u-boot_%.bbappend
@@ -6,7 +6,7 @@ SRC_URI:append:rpi = " \
 
 SRC_URI:append:rpi = " file://0001-rpi-always-set-fdt_addr-with-firmware-provided-FDT-address.patch"
 
-DEPENDS:append:rpi = " u-boot-default-script"
+DEPENDS:append:rpi = " ${@oe.utils.conditional('UBOOT_ENV', '', 'u-boot-default-script', '', d)}"
 
 do_install:append:rpi () {
     install -d ${D}${sysconfdir}
-- 
2.34.1

